### PR TITLE
Auto create issue on install failure

### PR DESCRIPTION
**Pull request by autoantwort**
Closes https://github.com/microsoft/vcpkg/discussions/17490

### CODE REVIEW

**Comment by autoantwort** [comment]
For example https://github.com/microsoft/vcpkg/issues/22206 and https://github.com/microsoft/vcpkg/issues/22207 were created by this

**Review by strega-nil-ms — COMMENTED** [review]
This is really cool! I want @ras0219-msft to take a look first (so it probably won't get merged until january), but I'm definitely a :+1:.

File: src/vcpkg/base/system.cpp [hunk 1]
```diff
@@ -367,6 +368,21 @@ namespace vcpkg
...
+#if defined(_WIN32)
+        return "windows";
+#elif defined(__APPLE__)
+        return "osx";
+#elif defined(__FreeBSD__)
+        return "freebsd";
+#elif defined(__OpenBSD__)
+        return "openbsd";
+#else
+        return "linux";
```
Comment (strega-nil-ms) [inline src/vcpkg/base/system.cpp:383]:
```suggestion
#elif defined(__linux__)
        return "linux";
#else
        return "unknown"
```

**Review by strega-nil-ms — CHANGES_REQUESTED** [review]
We shouldn't assume linux, since someone might be running this on z/OS for example.

**Comment by autoantwort** [comment]
Can be merged now :)

**Review by BillyONeal — APPROVED** [review]

**Pull request merged**